### ISSUE TITLE
GPL Dependency

### ISSUE THREAD

**Issue by martin-traverse** [flyway/flyway#3565]
In version 9.3.1, flyway-core introduces a dependency on com.electronwill.night-config:toml. This is a GPL (v3) dependency, which contaminates the license dependency tree. Our project is licensed as Apache 2 and part of the FINOS foundation for open source projects in financial services, so we absolutely cannot introduce GPL dependencies. Other open source projects releasing under a permissive license will have similar issues due to license incompatibility.

Is this a dependency you can remove and replace in a future version? For now we have stuck on 9.3, but our compliance requirements mean we cannot stick on old versions for very long. If this is something you can fix soon we can stay on 9.3 until the fix comes, otherwise sadly we'll have to look at using another tool, which would be a shame because Flyway is a great fit for us and really easy to use.

Last good version POM: https://search.maven.org/artifact/org.flywaydb/flyway-core/9.3.0/jar
First bad version POM: https://search.maven.org/artifact/org.flywaydb/flyway-core/9.3.1/jar

Any info will be most helpful!

###### Which version and edition of Flyway are you using?

9.3.1 and later has the issue, so we are using 9.3.0!

###### If this is not the latest version, can you reproduce the issue with the latest one as well? (Many bugs are fixed in newer releases and upgrading will often resolve the issue)

Issue still exists in 9.7.0

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)

N/A

###### Which database are you using? (Type & version)

N/A

###### Which operating system are you using?

N/A

###### What did you do? (Please include the content causing the issue, any relevant configuration settings, the SQL statement(s) that failed (if any), and the command you ran)

Updated to latest stable

###### What did you expect to see?

Expected license compliance checks to pass

###### What did you see instead?

License compliance checks fail, because of a GPL dependency from com.electronwill.night-config:toml

**Comment by tjs145** [comment]
The license for `com.electronwill.night-config` is only LGPL and we don't modify the source code for it, so it shouldn't affect the license of anything that depends on it. Either way, there are plans to move from `com.electronwill.night-config` to `jackson-dataformat` in the near future.

**Comment by ujhazib** [comment]
@martin-traverse sorry to ping you here, but can I ask the direct concern of yours about LGPLv3? Management levels above me also concerned here about the usage, but to be honest I read the license a couple of times and I don't see how it would enforce you to do ANY of the following
- make your sources open
- put your product under LGPLv3.

I think only these are the concerns a company would have against a license, but that's GPL and not LGPL where you would have to do such things. I spent a lot of time on checking licenses to be used for some small thing 10 years ago and even that time I was "aware" that one license which is not problematic is LGPL as long as I don't make a modification and I use library linking. I want to understand small details which might matter here, but I don't get too much feedback from legal anyways what is their problem with the licensing... I tried to search people who had similar concerns and get an answer to see another point of view.

And sorry to spam this issue, not gonna reopen it or whatsoever.
